### PR TITLE
Docs: Normalise Argument Clinic advanced topics headings

### PR DESCRIPTION
**Pull request by erlend-aasland**
----
:books: Documentation preview :books:: https://cpython-previews--106842.org.readthedocs.build/

### CODE REVIEW

File: Doc/howto/clinic.rst [hunk 1]
```diff
@@ -626,8 +623,8 @@ array) would be ``file``, but the C variable would be named ``file_obj``.
 You can use this to rename the ``self`` parameter too!
 
 
-Converting functions using PyArg_UnpackTuple
---------------------------------------------
+How to convert functions using PyArg_UnpackTuple
+------------------------------------------------
```
Comment (ezio-melotti) [inline Doc/howto/clinic.rst]:
```suggestion
How to convert functions using ``PyArg_UnpackTuple``
----------------------------------------------------
```

Should this have some markup?

File: Doc/howto/clinic.rst [hunk 2]
```diff
@@ -903,17 +900,17 @@ it accepts, along with the default value for each parameter.
 Just run ``Tools/clinic/clinic.py --converters`` to see the full list.
 
 
-Py_buffer
----------
+How to use the Py_buffer converter
+----------------------------------
```
Comment (ezio-melotti) [inline Doc/howto/clinic.rst]:
```suggestion
How to use the ``Py_buffer`` converter
--------------------------------------
```
Same here.

File: Doc/howto/clinic.rst [hunk 3]
```diff
@@ -944,8 +941,8 @@ hard-coded encoding strings for parameters whose format units start with ``e``.
 
 .. _default_values:
 
-Parameter default values
-------------------------
+How to assign parameter default values
+--------------------------------------
```
Comment (ezio-melotti) [inline Doc/howto/clinic.rst]:
```suggestion
How to assign default values to parameter
-----------------------------------------
```
This sounds a bit better to me.

File: Doc/howto/clinic.rst [hunk 4]
```diff
@@ -979,8 +976,8 @@ behaves like a default value of ``None``, but the C variable is initialized
 with ``NULL``.
 
 
-Expressions specified as default values
----------------------------------------
+Who to use expressions as default values
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (ezio-melotti) [inline Doc/howto/clinic.rst]:
```suggestion
How to use expressions as default values
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: Doc/howto/clinic.rst [hunk 5]
```diff
@@ -1400,8 +1397,8 @@ specifically the implementation of ``CReturnConverter`` and
 all its subclasses.
 
 
-METH_O and METH_NOARGS
-----------------------
+How to convert METH_O and METH_NOARGS functions
+-----------------------------------------------
```
Comment (ezio-melotti) [inline Doc/howto/clinic.rst]:
```suggestion
How to convert ``METH_O`` and ``METH_NOARGS`` functions
-------------------------------------------------------
```

File: Doc/howto/clinic.rst [hunk 6]
```diff
@@ -1422,8 +1419,8 @@ You can still use a self converter, a return converter, and specify
 a ``type`` argument to the object converter for ``METH_O``.
 
 
-tp_new and tp_init functions
-----------------------------
+How to convert tp_new and tp_init functions
+-------------------------------------------
```
Comment (ezio-melotti) [inline Doc/howto/clinic.rst]:
```suggestion
How to convert ``tp_new`` and ``tp_init`` functions
---------------------------------------------------
```

File: Doc/howto/clinic.rst (hunk 3 above)
Comment (erlend-aasland) [reply Doc/howto/clinic.rst]: Yeah, that's better; thanks!

File: Doc/howto/clinic.rst (hunk 1 above)
Comment (erlend-aasland) [reply Doc/howto/clinic.rst]: Yeah, that makes sense (also for the other cases you pointed out). We _could_ use C function markup here, but I'm not sure it matters that much in a heading.

**Comment by erlend-aasland** [comment]
Thanks for the improvements, Ezio!

**Pull request merged**